src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/StackExchangeRedis2Plus/SessionCache.cs

```diff
@@ -45,15 +49,14 @@ public void Harvest(ISegment hostSegment)
                 return;
```

Comment:
I think that we need to have a lock around this entire conditional. Without the lock, it's possible that the Cleanup method and the method to get a profiling session will be checking the state of things while this code decides to remove harvest the segment.